src/Stratis.Bitcoin/Base/ChainRepository.cs

```diff
@@ -17,23 +18,110 @@ public interface IChainRepository : IDisposable
         Task SaveAsync(ConcurrentChain chain);
     }
 
-    public class ChainRepository : IChainRepository
+    public interface IFinalizedBlockHeight : IDisposable
     {
+        /// <summary>Gets the finalized block height.</summary>
+        /// <returns>Height of a block that can't be reorged away from.</returns>
+        int GetFinalizedBlockHeight();
+
+        /// <summary>Loads the finalized block height.</summary>
```

Comment:
Loads the finalised block height from the database.

Reply:
Fixed